packages/ui/src/components/ValidatorList/KeystoresDataGrid.tsx

```diff
@@ -61,6 +69,106 @@ export default function KeystoresDataGrid({
+  
+    // Only proceed if there is at least one row with an index
+    // new validatos will not have an index yet
+    if (rowsWithIndex.length > 0) {
+      try {
+        // Initialize an array to hold the batches
+      const batches = [];
+      // Create a helper array from rowsWithIndex to avoid modifying the original array
+      const helperArray = [...rowsWithIndex];
+
+      // Split the indices into batches of 100 or less
+      while (helperArray.length) {
+        batches.push(helperArray.splice(0, 100));
+      }
+        for (const batch of batches) {
+          const apiUrl =
+            network === "mainnet" ? MAINNET_ORACLE_URL : TESTNET_ORACLE_URL;
+
+          // Call the Oracle API to get the subscription status of the validators in the batch by their index
+          const response = await fetch(
+            `${apiUrl}/memory/validatorsbyindex/${batch
+              .map((row) => row.index)
+              .join()}`
+          );
+
+          const data = await response.json();
+
+          // Healthy checks: check that the response is valid and that the response is ok
+          if (!isValidOracleResponse(data)) {
+            throw new Error(
+              "Unexpected response structure when calling Oracle! Could not fetch validator subscription status"
+            );
+          }
+          if (!response.ok) {
+            throw new Error(`HTTP error when calling Oracle! ${response}`);
+          }
+
+          // If we get here, we assume the response is ok and process the data
+          const foundValidators = data.found_validators || []; //treat array as empty if undefined to avoid errors
+          for (const foundValidator of foundValidators) {
+            newValidatorSubscriptionStatus[foundValidator.validator_key] =
+              foundValidator.status;
+          }
+          // process not found validators and mark them as unsubscribed
+          const notFoundValidators = data.not_found_validators || []; //treat array as empty if undefined to avoid errors
+          for (const notFoundValidator of notFoundValidators) {
+            const notFoundValiToString = notFoundValidator.toString();
+
+            // find the validator in the batch and mark it as not subscribed
+            const validator = batch.find(
+              (validator) => validator.index === notFoundValiToString
+            );
+            if (validator) {
+              newValidatorSubscriptionStatus[validator.pubkey] =
+                MevSpSubscriptionStatus.NOT_SUBSCRIBED;
+            }
+          }
+        }
+        //Update the object containing the subscription status by pubkey of all validators
+        setValidatorsData(newValidatorSubscriptionStatus);
+        setOracleCallError(undefined);
+      } catch (e) {
+        setOracleCallError(e.message);
+      }
+    } else {
+      setOracleCallError(
+        "Skipping Oracle subscription status fetch, no validator index could be fetch. Is your consensus client synced?"
```

Comment:
It might be implemented in a separated PR, but, can we know in advanced if is synced or not? this way we would display useful data instead of an error with a "maybe"

Reply:
True! Didnt, think about it, oracle has an endpoint that tells you if it is synced. I have implemented it in code now, just before trying to call subscription status (if not sync, subscription status call is skipped)